### DISCUSSION TITLE
Generated prisma util types causing model name clash

### DISCUSSION THREAD

**Discussion by vuki656** [prisma/prisma#13007] [Q&A] [answered]
We have a table called `checks` and would use the Prisma model `Check` this would result in Prisma generating types for that model including a select type called `Prisma.CheckSelect`. Unfortunately, they already have an internal function called `Prisma.CheckSelect` which clashes with our name.

The workaround was to name the model in the Prisma schema `Check_` in order to avoid the name clash. 

I feel like the exposed helper types like `Prisma.CheckSelect` should be prefixed by something in order to avoid the name clash.

Is this something that would potentially be resolved on the Prisma side or am I going to have to live with the `Prisma.Check_Select` ?

**Comment by austincrim** [comment, marked as answer]
Hey @vuki656 👋 ,

We are aware of these issues with Prisma's generated types. We are tracking them in an overarching issue [here](https://github.com/prisma/prisma/issues/12469). This will be taken care of in a future version of Prisma after we prioritize it and design a solution.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
